apps/www/src/components/plate-ui/playground-turn-into-dropdown-menu.tsx

```diff
@@ -109,20 +108,22 @@ const defaultItem = items.find((item) => item.value === ELEMENT_PARAGRAPH)!;
 
 export function PlaygroundTurnIntoDropdownMenu(props: DropdownMenuProps) {
   const value: string = useEditorSelector((editor) => {
-    if (!isSelectionExpanded(editor)) {
-      const entry = findNode<TElement>(editor!, {
-        match: (n) => isBlock(editor, n),
-      });
+    let commonSelection: string | undefined;
+    const codeBlockEntries = getNodeEntries(editor, {
+      match: (n) => isBlock(editor, n),
+    });
+    const nodes = Array.from(codeBlockEntries);
+    nodes.forEach(([node]) => {
+      const type: string = (node?.type as string) || ELEMENT_PARAGRAPH;
 
-      if (entry) {
-        return (
-          items.find((item) => item.value === entry[0].type)?.value ??
-          ELEMENT_PARAGRAPH
-        );
+      if (!isDefined(commonSelection)) {
+        commonSelection = type;
+      } else if (commonSelection !== type) {
+        commonSelection = undefined;
       }
```

Comment:
Thanks for the PR! It looks like this logic might cause the `commonSelection` variable to flip between `undefined` and some value. For example, if the types we get are `['typeA', 'typeB', 'typeC', 'typeD', 'typeE']`, the value of `commonSelection` after each step is `['typeA', undefined, 'typeC', undefined, 'typeE']`.

Reply:
Thanks @12joan will improve this logic which will prevent flip with below logic 

![Screenshot 2024-06-05 at 12 51 44 AM](https://github.com/udecode/plate/assets/66991625/bf4c9e24-4fa0-482e-ae97-c23b6f5ac6f1)

Reply:
This has the same problem.

```
type    commonSelection     uniqueTypeFound
------------------------------------------
N/A     undefined           false
typeA   typeA               true
typeB   undefined           false
typeC   typeC               true
typeD   undefined           false
typeE   typeE               true
```

Maybe get the type of the first node and then use `nodes.every` to check if they all match?

Reply:
![Screenshot 2024-06-05 at 2 05 16 AM](https://github.com/udecode/plate/assets/66991625/0a745d28-7ae9-4756-b928-c5c51dddfbb8)